src/main/java/com/fouristhenumber/utilitiesinexcess/common/items/ItemInversionSigilActive.java

```diff
@@ -2,29 +2,527 @@
+        ItemStack[] CHEST_EAST_CONTENTS = new ItemStack[62];
+        for (int i = 0; i < 31; i++) {
+            CHEST_EAST_CONTENTS[2 * i] = new ItemStack(Items.potionitem, 1, POTION_IDS[i]);
+            CHEST_EAST_CONTENTS[2 * i + 1] = new ItemStack(Items.potionitem, 1, POTION_IDS[i] + 8192);
+        }
+
+        ItemStack[] CHEST_SOUTH_CONTENTS = { new ItemStack(Blocks.grass), new ItemStack(Blocks.lapis_ore),
+            new ItemStack(Blocks.dirt), new ItemStack(Blocks.obsidian), new ItemStack(Blocks.sand),
+            new ItemStack(Blocks.diamond_ore), new ItemStack(Blocks.gravel), new ItemStack(Blocks.redstone_ore),
+            new ItemStack(Blocks.gold_ore), new ItemStack(Blocks.clay), new ItemStack(Blocks.iron_ore),
+            new ItemStack(Blocks.emerald_ore), new ItemStack(Blocks.coal_ore) };
+
+        ItemStack[] CHEST_WEST_CONTENTS = { new ItemStack(Items.record_13), new ItemStack(Items.record_mellohi),
+            new ItemStack(Items.record_cat), new ItemStack(Items.record_stal), new ItemStack(Items.record_blocks),
+            new ItemStack(Items.record_strad), new ItemStack(Items.record_chirp), new ItemStack(Items.record_ward),
+            new ItemStack(Items.record_far), new ItemStack(Items.record_11), new ItemStack(Items.record_mall),
+            new ItemStack(Items.record_wait) };
+
+        if (world.getTileEntity(beaconX, beaconY, beaconZ - 5) instanceof TileEntityChest chest) {
+            chestNorthContentsOk = checkChest(chest, CHEST_NORTH_CONTENTS, InversionConfig.northChestRequiredItems);
+        } else {
+            chestNorthContentsOk = false;
+        }
+        if (world.getTileEntity(beaconX + 5, beaconY, beaconZ) instanceof TileEntityChest chest) {
+            chestEastContentsOk = checkChest(chest, CHEST_EAST_CONTENTS, InversionConfig.eastChestRequiredItems);
+        } else {
+            chestEastContentsOk = false;
+        }
+        if (world.getTileEntity(beaconX, beaconY, beaconZ + 5) instanceof TileEntityChest chest) {
+            chestSouthContentsOk = checkChest(chest, CHEST_SOUTH_CONTENTS, InversionConfig.southChestRequiredItems);
+        } else {
+            chestSouthContentsOk = false;
+        }
+        if (world.getTileEntity(beaconX - 5, beaconY, beaconZ) instanceof TileEntityChest chest) {
+            chestWestContentsOk = checkChest(chest, CHEST_WEST_CONTENTS, InversionConfig.westChestRequiredItems);
+        } else {
+            chestWestContentsOk = false;
+        }
+
+        if (!(dimensionOk && spiralOk
+            && chestNorthContentsOk
+            && chestEastContentsOk
+            && chestSouthContentsOk
+            && chestWestContentsOk)) {
+            return;
+        }
+
+        // Ritual has now succeeded
+
+        if (!world.isRemote && !source.siege) {
+            player.addChatMessage(new ChatComponentTranslation("chat.pseudo_inversion_ritual.complete"));
+        }
+        if (!source.siege) {
```

Comment:
and this condition is now the same as the condition above, you can combine the code